src/all/mangaball/src/eu/kanade/tachiyomi/extension/all/mangaball/MangaBall.kt

```diff
@@ -0,0 +1,409 @@
+package eu.kanade.tachiyomi.extension.all.mangaball
+
+import android.util.Log
+import androidx.preference.PreferenceScreen
+import androidx.preference.SwitchPreferenceCompat
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.POST
+import eu.kanade.tachiyomi.source.ConfigurableSource
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.HttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import keiyoushi.utils.firstInstance
+import keiyoushi.utils.getPreferencesLazy
+import keiyoushi.utils.parseAs
+import keiyoushi.utils.tryParse
+import okhttp3.Call
+import okhttp3.Callback
+import okhttp3.FormBody
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import okhttp3.Response
+import okhttp3.internal.closeQuietly
+import okio.IOException
+import org.jsoup.nodes.Document
+import rx.Observable
+import java.lang.UnsupportedOperationException
+import java.text.DecimalFormat
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+class MangaBall(
+    override val lang: String,
+    private vararg val siteLang: String,
+) : HttpSource(), ConfigurableSource {
+
+    override val name = "Manga Ball"
+    override val baseUrl = "https://mangaball.net"
+    override val supportsLatest = true
+    private val preferences by getPreferencesLazy()
+
+    override val client = network.cloudflareClient.newBuilder()
+        .addInterceptor { chain ->
+            var request = chain.request()
+            if (request.url.encodedPath.contains("/api/")) {
```

Comment:
Why mix `contains` and `pathSegments` (as seen in `deepLink`)? Should be consistent unless there's a non-obvious reason for it.